### PR TITLE
ST-1060-ccd-case-document-am-api integration for preview

### PR DESCRIPTION
**Pull request by geciciserkan**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ST-1060

https://tools.hmcts.net/jira/browse/SPTRIBS-

**Before merging a pull request make sure that:**

- [ ] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

**Review by muradalamv1 — DISMISSED** [review]

**Review by osheaf01 — APPROVED** [review]
LGTM

**Review by muradalamv1 — DISMISSED** [review]

**Review by spotstory — APPROVED** [review]

File: charts/sptribs-case-api/values.preview.template.yaml [hunk 1]
```diff
@@ -6,8 +6,8 @@ java:
   image: ${IMAGE_NAME}
   ingressHost: ${SERVICE_FQDN}
   environment:
-    CASE_DATA_STORE_BASEURL: http://{{ .Release.Name }}-ccd-data-store-api
-    CASE_DOCUMENT_AM_BASEURL: http://ccd-case-document-am-api-aat.service.core-compute-aat.internal
+    CASE_DATA_STORE_BASEURL: http://ccd-data-store-api-sptribs-case-api-pr-${CHANGE_ID}.service.core-compute-preview.internal
```
Comment (mattnayler) [inline charts/sptribs-case-api/values.preview.template.yaml:9]: I would suggest that for your chart file your preview chart file you go with one naming convention only for those calls you know are _preview_ to _preview_.

so: hopefully 

```
CASE_DATA_STORE_BASEURL: http://${SERVICE_NAME}-ccd-data-store-api
```

or

```
CASE_DATA_STORE_BASEURL: http://sptribs-case-api-pr-${CHANGE_ID}-ccd-data-store-api
```

(I think/hope the above are both equivalent: so pick the one that is most readable to your team).

and follow the same throughout for all that you know are preview-preview connections.  (i.e. anything that is not already configured as AAT). and look to leave out: `.service.core-compute-preview.internal` until you are configuring the FTA scripts inside jenkins config.

File: charts/sptribs-case-api/values.preview.template.yaml [hunk 2]
```diff
@@ -92,7 +92,7 @@ ccd:
     java:
       disableKeyVaults: true
       environment:
-        CCD_DATA_STORE_URL: http://${SERVICE_NAME}-ccd-data-store-api
+        CCD_DATA_STORE_URL: http://sptribs-case-api-pr-${CHANGE_ID}-ccd-data-store-api
```
Comment (mattnayler) [inline charts/sptribs-case-api/values.preview.template.yaml:95]: NB: the lines above appear wrong to me: please re-review

```
  am-role-assignment-service:
      ingressHost: ccd-data-store-api-${SERVICE_FQDN}
  am-role-assignment-service:
    java:
      disableKeyVaults: true
```

... the ingressHost should probably be an `am-role-assignment-service` specific one.

**Review by mattnayler — COMMENTED** [review]
Partial review for @geciciserkan

File: charts/sptribs-case-api/values.preview.template.yaml [hunk 3]
```diff
@@ -6,8 +6,8 @@ java:
   image: ${IMAGE_NAME}
   ingressHost: ${SERVICE_FQDN}
   environment:
-    CASE_DATA_STORE_BASEURL: http://{{ .Release.Name }}-ccd-data-store-api
-    CASE_DOCUMENT_AM_BASEURL: http://ccd-case-document-am-api-aat.service.core-compute-aat.internal
+    CASE_DATA_STORE_BASEURL: http://${SERVICE_NAME}-ccd-data-store-api
+    CASE_DOCUMENT_AM_BASEURL: http://ccd-case-document-am-api-sptribs-case-api-pr-${CHANGE_ID}.service.core-compute-preview.internal
```
Comment (mattnayler) [inline charts/sptribs-case-api/values.preview.template.yaml:10]: http://${SERVICE_NAME}-ccd-case-document-am-api

**Review by ElifDenizEsea — APPROVED** [review]

**Pull request merged**